lms/resources/_js_config/__init__.py

```diff
@@ -71,13 +71,20 @@ def add_document_url(self, document_url):
             }
         elif document_url.startswith("vitalsource://"):
             vitalsource_svc = self._request.find_service(name="vitalsource")
-            launch_url, launch_params = vitalsource_svc.get_launch_params(
-                document_url, self._request.lti_user
-            )
-            self._config["vitalSource"] = {
-                "launchUrl": launch_url,
-                "launchParams": launch_params,
-            }
+            if self._request.feature("vitalsource_anon_launch"):
+                self._config["vitalSource"] = {
+                    "launchUrl": vitalsource_svc.get_iframe_url(document_url),
+                    "launchParams": None,
```

Comment:
You can just omit `launchParams` entirely here. It was present in my earlier code to avoid an extra branch. The frontend only cares whether it is nullish or not.